src/Spectre.Service/Loaders/DatasetLoader.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*
+ * DatasetLoader.cs
+ * Class loading datasets from either local root or remote root.
+ *
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using Ninject;
+using Spectre.Data.Datasets;
+using Spectre.Service.Configuration;
+
+namespace Spectre.Service.Loaders
+{
+    /// <summary>
+    /// Class for loading datasets from specified directories.
+    /// </summary>
+    public class DatasetLoader
+    {
+        #region Fields
+
+        /// <summary>
+        /// Root for local directory.
+        /// </summary>
+        private readonly string _localRoot;
+
+        /// <summary>
+        /// Root for remote directory.
+        /// </summary>
+        private readonly string _remoteRoot;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DatasetLoader"/> class.
+        /// </summary>
+        /// <param name="dataRootConfig">Validated configuration of needed directories.</param>
+        public DatasetLoader(DataRootConfig dataRootConfig)
+        {
+            _localRoot = dataRootConfig.LocalPath;
+            _remoteRoot = dataRootConfig.RemotePath;
+        }
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Handle to file system.
+        /// </summary>
+        [Inject]
+        public IFileSystem FileSystem { private get; set; }
+
+        #endregion
+
+        #region Loaders
+
+        /// <summary>
+        /// Returns dataset basing directly on the name of the file.
+        /// </summary>
+        /// <param name="name">File name.</param>
+        /// <returns>Found dataset.</returns>
+        /// <exception cref="FileNotFoundException">Throws when the file is not found both locally and remotely.</exception>
+        /// <exception cref="DatasetLoadException">Throws when the loader fails to create dataset from the file.</exception>
+        public IDataset GetFromName(string name)
+        {
+            string fullPathLocal = Path.Combine(_localRoot, name);
+
+            if (!FileSystem.File.Exists(fullPathLocal))
+            {
+                string fullPathRemote = Path.Combine(_remoteRoot, name);
+                if (!FileSystem.File.Exists(fullPathRemote))
+                {
+                    throw new FileNotFoundException("File not found neither locally nor remotely.", name);
```

Comment:
What about `DatasetNotFoundException`?

Reply:
For this purpose is such specification neccessary?

Reply:
What I have in mind are logs, which we will enable in future. Usually, the exception name is saved, sometimes with message, call stack, etc. The more precise we are here, the better we will be able to debug from the logs. What will tell us `FileNotFoundException` for 2-file `imzml` file? Is this `.imzml` file missing or `.ibd`? One of these will be our first guess, while the case is that a dataset does not exist at all.